### PR TITLE
Basic support for macros

### PR DESCRIPTION
**Pull request by w1th0utnam3**
What do you think about this for basic macros support? (issue #1) I'm not sure if this is an idiomatic solution because I just started looking into syn.

I think it might be sensible to somehow make this behavior optional, by either
- adding a flag to the proc macros to enable macro support
- adding another proc macro family that has macro support, with the current macros not supporting it
- hiding it behind a crate wide flag

### CODE REVIEW

**Comment by Andlon** [comment]
Thanks, this is great! I'll try to review this later today, or if not then at least in the next few days.

As for toggling this behavior, I like your first suggestion the best. Something like

```rust
#[replace_float_literals(T::from(literal), visit_macros=false)]`
```
or something to that effect, with the default being to actually visit all macros (I would suspect this is usually what one wants). I'm not sure exactly what form/syntax is best here. More generally, it would be nice to exclude specific macros, e.g. something like `exclude_macros=[vec, array]` or something like this. I think that this would better belong in a separate issue/PR though.

If you don't mind, it would be nice to already include this toggle behavior before we get this merged. I'd like to hear your own opinion on what you think is the best way forward here.

File: tests/tests.rs
Comment (Andlon) [inline tests/tests.rs]: Is there a reason for not just having a function without a return value here?

File: src/lib.rs
Comment (Andlon) [inline src/lib.rs]: minor typo in function name

**Review by Andlon — CHANGES_REQUESTED** [review]
Looks good to me, just discovered a typo. Otherwise I'll just wait for your feedback on how to proceed with the optional toggle for macros (an alternative would of course be to just merge directly, which I think is reasonable too).

**Comment by w1th0utnam3** [comment]
Yeah, I like your suggestion of an optional toggle to turn it off. I'll try to implement it before you merge it. I just didn't have the time to do it until now.
Before a new release we'll also have to document the exact behaviour and what macros are supported in the readme.

**Comment by Andlon** [comment]
Yes, good point about docs. Take your time, and let me know if I can help!

File: tests/tests.rs
Comment (w1th0utnam3) [reply tests/tests.rs]: No, I guess returning something here is completely  unnecessary.

Comment (w1th0utnam3) [reply tests/tests.rs]: Fixed

File: src/lib.rs
Comment (w1th0utnam3) [reply src/lib.rs]: Fixed

**Review by Andlon — APPROVED** [review]

**Review by Andlon — APPROVED** [review]

**Pull request merged**
